### PR TITLE
[Snyk] Upgrade eslint-config-expensify from 2.0.16 to 2.0.44

### PR DESCRIPTION
**Pull request by MelvinBot**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to upgrade eslint-config-expensify from 2.0.16 to 2.0.44.</h3>

As this is a private repository, Snyk-bot does not have access. Therefore, this PR has been created automatically, but appears to have been created by a real user.</br>
:information_source: Keep your dependencies up-to-date. This makes it easier to fix existing vulnerabilities and to more quickly identify and fix newly disclosed vulnerabilities when they affect your project.
<hr/>

- The recommended version is **22 versions** ahead of your current version.
- The recommended version was released **a month ago**, on 2024-02-15.

The recommended version fixes:

Severity                   | Issue                | PriorityScore (*)                 | Exploit Maturity |
:-------------------------:|:-------------------------|-------------------------|:-------------------------
  | Prototype Pollution<br/> [SNYK-JS-MINIMIST-2429795](https://snyk.io/vuln/SNYK-JS-MINIMIST-2429795) | **292/1000**  <br/> **Why?** Proof of Concept exploit, CVSS 3.7  | Proof of Concept 

(*) Note that the real score may have changed since the PR was raised.

<details>
<summary><b>Release notes</b></summary>
<br/>
  <details>
    <summary>Package name: <b>eslint-config-expensify</b></summary>
    <ul>
      <li>
        <b>2.0.44</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.44">2024-02-15</a></br><p>2.0.44</p>
      </li>
      <li>
        <b>2.0.43</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.43">2023-12-23</a></br><p>2.0.43</p>
      </li>
      <li>
        <b>2.0.42</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.42">2023-10-24</a></br><p>2.0.42</p>
      </li>
      <li>
        <b>2.0.41</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.41">2023-10-18</a></br><p>2.0.41</p>
      </li>
      <li>
        <b>2.0.39</b> - 2023-09-21
      </li>
      <li>
        <b>2.0.38</b> - 2023-06-13
      </li>
      <li>
        <b>2.0.36</b> - 2023-04-26
      </li>
      <li>
        <b>2.0.35</b> - 2023-04-25
      </li>
      <li>
        <b>2.0.33</b> - 2023-04-06
      </li>
      <li>
        <b>2.0.32</b> - 2023-04-05
      </li>
      <li>
        <b>2.0.31</b> - 2023-02-17
      </li>
      <li>
        <b>2.0.30</b> - 2022-09-05
      </li>
      <li>
        <b>2.0.29</b> - 2022-06-07
      </li>
      <li>
        <b>2.0.27</b> - 2022-05-27
      </li>
      <li>
        <b>2.0.24</b> - 2022-03-10
      </li>
      <li>
        <b>2.0.23</b> - 2022-01-28
      </li>
      <li>
        <b>2.0.22</b> - 2022-01-21
      </li>
      <li>
        <b>2.0.21</b> - 2021-12-15
      </li>
      <li>
        <b>2.0.20</b> - 2021-12-03
      </li>
      <li>
        <b>2.0.19</b> - 2021-11-24
      </li>
      <li>
        <b>2.0.18</b> - 2021-11-08
      </li>
      <li>
        <b>2.0.17</b> - 2021-11-05
      </li>
      <li>
        <b>2.0.16</b> - 2021-08-12
      </li>
    </ul>
    from <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases">eslint-config-expensify GitHub release notes</a>
  </details>
</details>

<details>
  <summary><b>Commit messages</b></summary>
  </br>
  <details>
    <summary>Package name: <b>eslint-config-expensify</b></summary>
    <ul>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/72d77f0eae7e66775b98ded3e9c7f7b0a6144dc4">72d77f0</a> Merge pull request #86 from Expensify/OSBotify-bump-version-9ba6309f-decf-418e-9315-d283acdcef96</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/c321591ba76d421e6ac5e61bf8717c18b9bf8030">c321591</a> 2.0.44</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/a15f930699c3149eee46c2df5391620ccc75195b">a15f930</a> Merge pull request #85 from blazejkustra/react-hooks/recommended</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/ec28d49b4383ae55c66981da8a421326c4519914">ec28d49</a> Downgrade some packages</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/94b5c77c5250ed141b47af6fbdd2791831111834">94b5c77</a> Downgrade jest</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/62d889dbd1b92ea5c04d32441339e18be1826a92">62d889d</a> Fix formatting</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/0b9c68f3b806ea097fe5f5cd01f49e6bd596e999">0b9c68f</a> Downgrade eslint to version used in NewDot</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/cc43cc006f76ef1bc4290844268e6101cf650c71">cc43cc0</a> Fix formatting</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/76ca522b6f3f39b9cb783e44cd4de3064042422d">76ca522</a> Add react-hooks/recommended</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/822601b25a838459293c291342ffe852b4b6bfc7">822601b</a> Merge pull request #84 from Expensify/OSBotify-bump-version-fa3ad8d2-c6b0-439c-afdb-89164e0f6505</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/4feab03c3e6c1c42ad6a5d37d75d37a2b103a00c">4feab03</a> 2.0.43</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/5155010f3410b9055dfcdf93a9dd2d4d6d7de78b">5155010</a> Merge pull request #83 from ishpaul777/lint/prohibit-default-props</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/f9a5a46369c3590df548a3c921d88364ff2cdfd6">f9a5a46</a> Merge branch &#x27;lint/prohibit-default-props&#x27; of github.com:ishpaul777/eslint-config-expensify into lint/prohibit-default-props</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/b516d74c30c36eee5d03997155c16aa53b041a36">b516d74</a> adds the lint rule</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/4b76fe8b0561baad7c498d2c81f214c07faafe4e">4b76fe8</a> adds the lint rule</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/17dbfa4e063bfe50e94ebbc615e9d7dcf99203b0">17dbfa4</a> Merge pull request #81 from Expensify/OSBotify-bump-version-f6e29528-a56b-48ae-a3ef-eaac19508aa7</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/0cf30ef8dfca185dfe18726b2304325d8b08a04a">0cf30ef</a> 2.0.42</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/3c5d7de74ef00d80231e8d16a3481d24feeb4162">3c5d7de</a> Merge pull request #76 from ikevin127/fix/28334</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/2d02e8f04dc77340eb675af91b3a0c734b163bcf">2d02e8f</a> removed display-name-property rule</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/49f110377c597defdf11fea76a1772f3539dae90">49f1103</a> Merge pull request #80 from Expensify/OSBotify-bump-version-6627c903-f930-4619-9d95-b4521320bd39</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/fd191dc7ff56652bd1817ea10dcf82b9d79a4e7e">fd191dc</a> 2.0.41</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/c86595f01e09c3c34165f27f7238f634fef7e0fc">c86595f</a> Merge pull request #79 from Expensify/cmartins-rmBuildScript</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/60ee1b71ebaf6e17a5c26ba8ade49398481c3b4c">60ee1b7</a> rm build script from publish workflow</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/15085ff946ff34e52c04d37721f9cfa5c01fe59c">15085ff</a> Merge pull request #78 from Expensify/OSBotify-bump-version-6d8a8e96-c93a-4e35-afd3-17d7f8103379</li>
    </ul>

   <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/compare/49063030f091e8a1d068eaffa7297997c944d132...72d77f0eae7e66775b98ded3e9c7f7b0a6144dc4">Compare</a>
  </details>
</details>
<hr/>

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open upgrade PRs.*

For more information:  

🧐 [View latest project report](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🛠 [Adjust upgrade PR settings](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72/settings/integration?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🔕 [Ignore this dependency or unsubscribe from future upgrade PRs](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72/settings/integration?pkg&#x3D;eslint-config-expensify&amp;utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr#auto-dep-upgrades)

### CODE REVIEW

**Review by lakchote — APPROVED** [review]

**Comment by blazejkustra** [comment]
Hi @lakchote, version 2.0.44 bumped some of internal dependencies in `expensify-common`. Some of these changes affected in changes in default ESlint rules. [here](https://github.com/Expensify/App/pull/36849) is a PR that bumped this verion in NewDot, which required some changes in package.json (bumping eslint related packages) and a lot of changes and little fixes throughout the codebase.

We did the same for react-native-onyx, but this required less changes.

**Comment by blazejkustra** [comment]
Error code from lint javascript looks quite odd, but looking at the eslint setup in this repo I think it needs some adjustements. For example eslint isn't even a dependency 🤔 

Taking a deeper look now.

**Comment by blazejkustra** [comment]
I think eslint setup is the problem:
```ts
npm i eslint@7.6.0 @lwc/eslint-plugin-lwc@latest --save-dev
```

After this, eslint works well but returns a lot of errors:
image

I think we should fix those errors and setup prettier in this repo. This should resolve all these issues

**Comment by lakchote** [comment]
Thanks for your reactivity!

> For example eslint isn't even a dependency 🤔
> 

That is _really_  surprising.

I've started to work on it, using `prettier` didn't fix all the errors but I'd say at least half of them.

I'll assign you for review when I'm done if you're okay @blazejkustra?

**Comment by blazejkustra** [comment]
Sure! Let me know once it's ready 😄

**Comment by lakchote** [comment]
@blazejkustra feel free to review, I'm looking at the license issue.

**Comment by blazejkustra** [comment]
I missed it, I'll have a look now

File: lib/str.js [hunk 1]
```diff
@@ -1127,4 +1118,5 @@ const Str = {
     },
 };
 
-export default Str;
+// eslint-disable-next-line import/prefer-default-export
+export { Str };
```
Comment (blazejkustra) [inline lib/str.js:1122]: Why not `export default Str`?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: lib/str.js (hunk 1 above)
Comment (blazejkustra) [reply lib/str.js:1122]: This could lead to a lot of changes in other repos such as NewDot, apart from that the PR looks good!

**Comment by blazejkustra** [comment]
Before merging I think we should try to update expensify-common in NewDot, to test these changes, wdyt? @lakchote

File: lib/str.js (hunk 1 above)
Comment (lakchote) [reply lib/str.js]: I don't know why but I thought I had an error if I didn't do it like this.

When retrying, I didn't get any anymore. It must have been when I was doing the refactoring, one edge case where I was in.

It's now corrected!

**Comment by lakchote** [comment]
> Before merging I think we should try to update expensify-common in NewDot, to test these changes, wdyt? @lakchote

This is a good idea to avoid potential regressions and bugs.

At first glance, it seems like it's working correctly. I've played around with markdown in chat conversations.

Could you try it yourself too?

I've modified `package.json` on my local env with (to save you time):

`"expensify-common": "git+ssh://git@github.com/Expensify/expensify-common.git#snyk-upgrade-f306335e3d63e056f0d5639e1a540f00",`

**Comment by lakchote** [comment]
@blazejkustra please prioritize this to save me time, otherwise I'll have to handle conflicts (there are other pull requests pending).

**Comment by blazejkustra** [comment]
@lakchote I'm so sorry, I forgot to add a comment, here! I testes it and it worked just fine. Let me double check in a moment

**Review by blazejkustra — APPROVED** [review]
@lakchote I installed it locally, tested live-markdown and some other functionalities. Typecheck and lint are working fine. LGTM!

**Review by lakchote — APPROVED** [review]
